src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/LdapDirectoryDAO.scala

```diff
@@ -329,4 +335,14 @@ class LdapDirectoryDAO(protected val ldapConnectionPool: LDAPConnectionPool, pro
     ldapConnectionPool.modify(petDn(petServiceAccount.id), modifications.asJava)
     petServiceAccount
   }
+
+  override def getManagedGroupAccessInstructions(groupName: WorkbenchGroupName): Future[Option[String]] = Future {
+    Option(ldapConnectionPool.getEntry(groupDn(groupName), Attr.accessInstructions)).map { entry =>
+      Option(entry.getAttributeValue(Attr.accessInstructions))
+    }.getOrElse(throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"$groupName not found")))
+  }
+
```

Comment:
```
for{
  entry <- Future{ldapConnectionPool.getEntry(groupDn(groupName), Attr.accessInstructions)}
  res <- Option(entry).fold(Future.failed(no group name found exception))(e => Option(e.getAttributeValue(Attr.accessInstructions)).fold(Future.failed(no attribute instruction exception))(x => Future.successful(x)))
} yield res
```
I'll do something like this. @dvoet might not like `fold`, but the main point is we'll know better where failure happens.

Reply:
I agree that I was confused by this block... but I am not less confused by this

I would try something like
```
    for {
      entry <- Future { ldapConnectionPool.getEntry(groupDn(groupName), Attr.accessInstructions) }
      res <- Option(entry).map {
        case None => Future.failed("no group name found exception")
        case Some(e) => Future.successful(Option(e.getAttributeValue(Attr.accessInstructions)))
    } yield res
```